app/code/Magento/Elasticsearch/SearchAdapter/Query/Builder/MatchQuery.php

```diff
@@ -95,26 +97,28 @@ public function build(array $selectQuery, RequestQueryInterface $requestQuery, $
     }
 
     /**
-     * Prepare query.
+     * Prepare query
      *
      * @param string $queryValue
      * @param string $conditionType
      * @return array
      */
-    protected function prepareQuery($queryValue, $conditionType)
+    private function prepareQuery(string $queryValue, string $conditionType): array
```

Comment:
Please revert this line, it's breaking compatibility

Reply:
Here is the reason why it's needed: https://github.com/magento/magento2/pull/33632#pullrequestreview-721876346